Scripts/XRLineRendererBase.cs

```diff
@@ -0,0 +1,283 @@
+﻿using System;
+using UnityEngine;
+
+/// <summary>
+/// A unified base class for the XR Line Renderer and XR Trail Renderer
+/// </summary>
+[RequireComponent(typeof(MeshRenderer))]
+[RequireComponent(typeof(MeshFilter))]
+[ExecuteInEditMode]
+public abstract class XRLineRendererBase : MonoBehaviour
```

Comment:
There's got to be a better name for this class. It seems weird to have an `XRLineRenderer` that inherits from `XRLineRendererBase`.